web/auth/auth.go

```diff
@@ -255,6 +261,30 @@ func newSession(c echo.Context, inst *instance.Instance, redirect *url.URL, long
 	return sessionID, nil
 }
 
+func migrateToHashedPassphrase(inst *instance.Instance, settings *settings.Settings, passphrase []byte) {
+	iterations := crypto.DefaultPBKDF2Iterations
+	salt := inst.PassphraseSalt()
+	pass, masterKey := crypto.HashPassWithPBKDF2(passphrase, salt, iterations)
+	hash, err := crypto.GenerateFromPassphrase(pass)
+	if err != nil {
+		inst.Logger().Errorf("Could not hash the passphrase: %s", err.Error())
+		return
+	}
+	inst.PassphraseHash = hash
+	settings.PassphraseKdfIterations = iterations
+	settings.PassphraseKdf = instance.PBKDF2_SHA256
+	if err := lifecycle.CreatePassphraseKey(settings, masterKey); err != nil {
+		inst.Logger().Errorf("Could not create passphrase key: %s", err.Error())
+		return
+	}
+	if err := couchdb.UpdateDoc(couchdb.GlobalDB, inst); err != nil {
+		inst.Logger().Errorf("Could not update: %s", err.Error())
+	}
+	if err := settings.Save(inst); err != nil {
```

Comment:
What happens if we could save the passphrase on the instance but not the KDF or the number of iterations on the settings?
Is there any way to create some sort of transaction around those 2?

Reply:
CouchDB has no transaction. If it happens, the user will have to reset its password.